src/Support/ListenerConfigurationChecker.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\EventDispatcher\Support;
+
+use Psr\Container\ContainerExceptionInterface;
+use Psr\Container\ContainerInterface;
+use function is_array;
+use function is_callable;
+use function is_string;
+
+final class ListenerConfigurationChecker
+{
+    private ContainerInterface $container;
+
+    public function __construct(ContainerInterface $container)
+    {
+        $this->container = $container;
+    }
+
+    /**
+     * Checks the given event configuration and throws an exception in some cases:
+     * - incorrect configuration format
+     * - incorrect listener format
+     * - listener is not a callable
+     * - listener is meant to be a method of an object which can't be instantiated
+     *
+     * @param array $configuration An array in format of [eventClassName => [listeners]]
+     *
+     * @psalm-suppress InvalidCatch
+     */
+    public function check(array $configuration): void
+    {
+        foreach ($configuration as $eventName => $listeners) {
+            if (!is_string($eventName)) {
+                throw new InvalidEventConfigurationFormatException(
+                    'Incorrect event listener format. Format with event name must be used.'
```

Comment:
```suggestion
                    "Incorrect event format.  '$eventName' must be event class name."
```

Reply:
Let's not use single quotes in the end-user text. Double quotes only.